src/explore-education-statistics-common/src/modules/table-tool/utils/mapTableHeadersConfig.ts

```diff
@@ -12,32 +12,44 @@ import {
  */
 export default function mapTableHeadersConfig(
   tableHeaders: UnmappedTableHeadersConfig,
-  meta: FullTableMeta,
+  table: FullTable,
 ): TableHeadersConfig {
+  const { subjectMeta, results } = table;
   const mapToFilters = (headers: TableHeader[]): Filter[] => {
-    return headers.map(header => {
+    // When terms are selected the time period range can include ones not displayed in the table,
+    // filter these out to ensure the reorder headings list is correct
+    const filteredHeaders = headers.filter(header => {
```

Comment:
Again, I think this can be slightly simplified to:

```
    const filteredHeaders = headers.filter(header => {
      if (header.type !== 'TimePeriod') {
           return true;
      }
      return results.find(result => result.timePeriod === header.value)
    });
```

or even:

```
    const filteredHeaders = headers.filter(header => 
        header.type !== 'TimePeriod') || 
        results.find(result => result.timePeriod === header.value));
```

to be super-succinct!  But maybe harder to follow!